MdeModulePkg/Bus/Pci/NvmExpressDxe/UnitTest/MediaSanitizeUnitTest.c

```diff
@@ -0,0 +1,1208 @@
+/** @file -- MediaSanitizeUnitTest.c
+  Placeholder/framework for developing a Media Sanitize unit test package.
+
+  Copyright (c) Microsoft Corporation.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#include <Uefi.h>
+#include <Library/BaseLib.h>
+#include <Library/BaseMemoryLib.h>
+#include <Library/UefiLib.h>
+#include <Library/DebugLib.h>
+#include <Library/UnitTestLib.h>
+#include <Library/UefiBootServicesTableLib.h>
+#include <Library/UefiApplicationEntryPoint.h>
+#include <Library/UefiRuntimeServicesTableLib.h>
+#include <Uefi/UefiBaseType.h>
+#include <Library/UefiBootManagerLib.h>
+#include <Protocol/NvmExpressPassthru.h>
+
+#include "../NvmExpress.h"
+#include "../NvmExpressBlockIo.h"
+#include "../NvmExpressMediaSanitize.h"
+#include "../NvmExpressHci.h"
+#include "../../../Include/Protocol/MediaSanitize.h"
+
+// #pragma warning ( disable : 4100 )
+
+EFI_STATUS
+EFIAPI
+NvmeDeviceUnitTestPassthru (
+  IN     EFI_NVM_EXPRESS_PASS_THRU_PROTOCOL        *This,
+  IN     UINT32                                    NamespaceId,
+  IN OUT EFI_NVM_EXPRESS_PASS_THRU_COMMAND_PACKET  *Packet,
+  IN     EFI_EVENT                                 Event OPTIONAL
+  )
+{
+  //
+  // Parse command packet for unit testing
+  //
+  // NVME_DEVICE_PRIVATE_DATA                  *Device;
+  EFI_NVM_EXPRESS_COMMAND     *Command;
+  EFI_NVM_EXPRESS_COMPLETION  *Completion;
+  EFI_STATUS                  Status;
+  NVME_CQ                     *Cqe;
+
+  DEBUG ((DEBUG_VERBOSE, "[          ] NvmeDeviceUnitTestPassthru: Enter\n"));
+
+  UT_ASSERT_NOT_NULL (This);
+  UT_ASSERT_NOT_NULL (Packet);

```

Comment:
this isn't a test function so not sure you should really use UT asserts in it. UT asserts are for test functions and the macros use the UT logging and error codes.  This function returns EFI_STATUS codes.

Reply:
Good point. I'll remove the UT_ASSERT...() calls.